startup/10-devices.py

```diff
@@ -183,20 +183,25 @@ def _update_readback_setpoint(self):
 
     def put(self, value):
         self._setpoint = float(value)
-
+        
         start_time = ttime.monotonic()
         getattr(atf_db, f"put_{self._dtype}")(
             atf_db.get_channel_index(
                 f"{self._db}::{self._psname};{self._write_suffix}"
             ),
             value,
         )
-        while ttime.monotonic() - start_time < self._timeout:
-            if abs(self._get_readback() - value) < self._tol:
-                break
-            else:
-                # not reached yet, wait a bit
-                ttime.sleep(0.1)
+        try:
+            if(ttime.monotonic() - start_time >= self._timeout):
+                raise Exception("Timeout")
```

Comment:
We can create a custom exception for this timeout instead of using a general `Exception`. That can be useful to catch the specific type of exception.

```py
class TimeoutException(Exception):
    ...
```
(defined at the top level)